.github/workflows/tests.yaml

```diff
@@ -44,15 +44,15 @@ jobs:
 
   numpy_compat_test:
     runs-on: ubuntu-latest
-    name: Build with Numpy 2.0.0, test with 1.23.5
+    name: Build and test with Numpy
```

Comment:
@EwoutH I'd like to keep this test as is. I know of some projects that can't update to numpy 2 yet 🙈 and so I'd like to assert that wheels built with the latest numpy work with the oldest supported numpy version.

Reply:
Should we have a matrix for this? On job for 1.x, another for 2.x?

Reply:
No need, 2.x will be used by default in every other test with Python >= 3.9.